pipelinewise/fastsync/commons/tap_postgres.py

```diff
@@ -274,6 +274,10 @@ def get_table_columns(self, table_name):
                     WHEN data_type = 'ARRAY' THEN 'array_to_json("' || column_name || '") AS ' || column_name
                     WHEN udt_name = 'time' THEN 'replace("' || column_name || E'"::varchar,\\\'24:00:00\\\',\\\'00:00:00\\\') AS ' || column_name
                     WHEN udt_name = 'timetz' THEN 'replace(("' || column_name || E'" at time zone \'\'UTC\'\')::time::varchar,\\\'24:00:00\\\',\\\'00:00:00\\\') AS ' || column_name
+                    WHEN udt_name in ('timestamp', 'timestamptz') THEN
+                       'CASE WHEN "' ||column_name|| E'" < \\'0001-01-01 00:00:00.000000\\' '
+                            'OR "' ||column_name|| E'" > \\'9999-12-31 23:59:59.999\\' THEN \\'9999-12-31 23:59:59.999\\' '
```

Comment:
To avoid some edge cases dates need be in milliseconds otherwise dates like `0000-01-01 00:00:00.000123` not converted to the fallback date in milliseconds.

Reply:
Thanks for the feedback!
How did you manage to insert `0000-01-01 00:00:00.000123` into the pg table tho? cuz I'm getting `date/time field value out of range:` when running `insert`

Reply:
I was only reading the code but didn't try it. I spotted the inconsistency of milliseconds vs microseconds and the potential edge case if the value in the db is `0000-01-01 00:00:00.000123`.

Maybe postgres doesn't allow it, but can we still use milliseconds to be mathematically correct? what do you think?

Reply:
update in https://github.com/transferwise/pipelinewise/pull/521/commits/d04465864347890749944f1983061d19435eacf8